Tests/UnitTests/Purchasing/Purchases/PurchasesLogInTests.swift

```diff
@@ -251,29 +251,43 @@ class ExistingUserPurchasesLogInTests: BasePurchasesLogInTests {
     }
 
     func testLogInClearsTrialEligibilityCache() {
-        expect(self.cachingTrialOrIntroPriceEligibilityChecker.invokedClearCache) == false
+        // set up updates customer info, which clears cache once. wait until it's done before checking the rest.
```

Comment:
these two tests weren't actually working - commenting out the logIn line would still work. This is because we're mocking the identity manager, which bypasses the customerInfomanager, which bypasses actually triggering a customer info update, which in turn skips the actual cache clearing we're trying to test.